docs/modules/ROOT/pages/guides/v3-migration/index.adoc

```diff
@@ -0,0 +1,159 @@
+[[v3-migration]]
+= 3.0.0 Migration
+This document lists all breaking changes from version 2.x.y to 3.0.0 and how to update.
+
+== How to upgrade
+Simply update `@neo4j/graphql` using npm or your package manager of choice:
+
+[source, bash, indent=0]
+----
+npm update @neo4j/graphql
+----
+
+== Async schema generation
+Schema generation is now asynchronous. Instead of using the property `schema`, now the method `getSchema` will return the schema
+as a _Promise_. This means that creating a server now requires awaiting for that method:
+
+Instead of
+[source, JavaScript, indent=0]
+----
+const neoSchema = new Neo4jGraphQL({ typeDefs, driver });
+const server = new ApolloServer({
+    schema: neoSchema.schema,
+});
+----
+
+Now you'll need to do the following:
+
+[source, JavaScript, indent=0]
+----
+const neoSchema = new Neo4jGraphQL({ typeDefs, driver });
+const schema = await neoSchema.getSchema();
+const server = new ApolloServer({
+    schema: schema
+});
+----
```

Comment:
Possibly also worth documenting this using `.then`? I imagine this will be appropriate for a lot of users.

```
----
const neoSchema = new Neo4jGraphQL({ typeDefs, driver });
neoSchema.getSchema().then((schema) => {
  const server = new ApolloServer({
    schema: schema
  });
});
----
```

Reply:
mmm, documenting both would be heavily redundant, so I guess it is promises vs async await

Thoughts?

Reply:
Does this come down to a matter a of (personal) preference? Then I vote for async await due to its slightly better readability IMO. But if the majority of our users are more familiar to promises then we should opt for that.

Reply:
I agree with you, although I think most of our examples are already riddled with async/await. Thoughts on this @darrellwarde @danstarns ?